added regex validation for names

simionescuraluca opened the pull request.

simionescuraluca committed:
added regex validation for names

dragosslujeru commented on src/main/java/com/vinyl/service/UserService.java:
```diff
@@ -1,8 +1,14 @@
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+//import javax.ws.rs.BadRequestException;
```
remove this line

simionescuraluca replied:
done

dragosslujeru commented on src/main/java/com/vinyl/controller/UserController.java:
```diff
@@ -28,6 +28,7 @@
 		user.setEmail(userDTO.getEmail());
 		user.setPass(userDTO.getPass());
 
+		
```
unintended change

simionescuraluca replied:
done

dragosslujeru commented on src/main/java/com/vinyl/service/UserService.java:
```diff
@@ -28,6 +34,14 @@ public User addUser(User user) {
 		user.setAddress("Defaul Address");
 		user.setRole(r);
 
+		Pattern p = Pattern.compile("^[A-Z][a-zA-Z]+$");
```
create a String constant for the pattern "^[A-Z][a-zA-Z]+$"

simionescuraluca replied:
done

dragosslujeru commented on src/main/java/com/vinyl/service/UserService.java:
```diff
@@ -28,6 +34,14 @@ public User addUser(User user) {
 		user.setAddress("Defaul Address");
 		user.setRole(r);
 
+		Pattern p = Pattern.compile("^[A-Z][a-zA-Z]+$");
```
move the validation to a new class that will handle only validation

simionescuraluca replied:
done

dragosslujeru commented on src/main/java/com/vinyl/service/exception/BadRequestException.java:
```diff
@@ -0,0 +1,14 @@
+
+@SuppressWarnings("serial")
+@ResponseStatus(value= HttpStatus.BAD_REQUEST)
+public class BadRequestException extends RuntimeException{
```
good job finding this solution

simionescuraluca replied:
thank you!

simionescuraluca committed:
resolved the comments on this task (formatting issues, created a
separate String for the regex expression and added an extra class which
handles only validation)

dragosslujeru commented on src/main/java/com/vinyl/service/Validation.java:
```diff
@@ -0,0 +1,23 @@
+
+public class Validation {
+
+	public static final String nameRegex = "^[A-Z][a-zA-Z]+$";
```
it is not used outside, make it private

dragosslujeru replied:
use constant naming conventions

simionescuraluca replied:
fixed

dragosslujeru commented on src/main/java/com/vinyl/service/Validation.java:
```diff
@@ -0,0 +1,23 @@
+public class Validation {
+
+	public static final String nameRegex = "^[A-Z][a-zA-Z]+$";
+	public static final Pattern VALID_NAME_REGEX = Pattern.compile(nameRegex);
```
it is a Pattern, not a regex VALID_NAME_PATTERN

simionescuraluca replied:
fixed

dragosslujeru commented on src/main/java/com/vinyl/service/Validation.java:
```diff
@@ -0,0 +1,23 @@
+import com.vinyl.model.User;
+import com.vinyl.service.exception.BadRequestException;
+
+public class Validation {
```
Validator

dragosslujeru replied:
make it a spring bean and inject into the service

simionescuraluca replied:
fixed

simionescuraluca committed:
intermediary commit

simionescuraluca committed:
Merge remote-tracking branch 'origin/master' into
New_Customer-Validate_first_name_and_last_name

Conflicts:
	src/main/java/com/vinyl/service/UserService.java

simionescuraluca committed:
validator bean injection + resolved other syntax comments

simionescuraluca committed:
@Component instead of @Bean
